plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -731,7 +730,7 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                     return Optional.of(ColumnMapping.longMapping(
                             TIMESTAMP_TZ_SECONDS,
                             shortTimestampWithTimeZoneReadFunction(),
-                            shortTimestampWithTimeZoneWriteFunction()));
+                            shortTimestampWithTimeZoneWriteFunction(column.getTimeZone())));
```

Comment:
> You can explicitly set a time zone for DateTime-type columns when creating a table. Example: DateTime('UTC'). If the time zone isn’t set, ClickHouse uses the value of the [timezone](https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#timezone) parameter in the server settings or the operating system settings at the moment of the ClickHouse server start.

So in this case should we always map DateTime to Timestamp with time zone ?

Reply:
Great question!
That will motivate me to reconsider my further thoughts!
However it goes far beyond this PR.

Reply:
Also a part of answer could be https://github.com/trinodb/trino/pull/23785#discussion_r1801398575